src/app/userInfo/page.tsx

```diff
@@ -0,0 +1,414 @@
+"use client";
+import React, { FormEvent, useState } from "react";
+import { useRouter } from "next/navigation";
+import Image from "next/image";
+import { format } from "date-fns";
+import { Calendar as CalendarIcon } from "lucide-react";
+
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { Label } from "@/components/ui/label";
+import { Textarea } from "@/components/ui/textarea";
+import { ModeToggle } from "@/components/ui/modeToggle";
+import { useToast } from "@/hooks/use-toast";
+import { Loader2, Upload } from "lucide-react";
+import { Calendar } from "@/components/ui/calendar";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { cn } from "@/lib/utils";
+import { updateUserInfo } from "@/service/user.service";
+
+interface FormData {
+  firstName: string;
+  lastName: string;
+  dateOfBirth: Date | undefined;
+  bio?: string;
+  profilePicture?: File | null;
```

Comment:
Why are we making profilePicture's type as File or null or undefined?
Are we treating null and undefined as separate states?